cesql/spec.md

```diff
@@ -353,6 +357,33 @@ left operand of the OR operation evalues to `true`, the right operand MUST NOT b
 
 #### 3.7. Type casting
 
+The following table indicates which type casts a CESQL engine MUST or MUST NOT support:
+
+| Type    | Integer  | String | Boolean  |
+| ------- | -------- | ------ | -------- |
+| Integer | N/A      | MUST   | MUST NOT |
+| String  | MUST     | N/A    | MUST     |
+| Boolean | MUST NOT | MUST   | N/A      |
```

Comment:
This is a valid choice, but I'm curious why you didn't choose to map `1` with `true` and `0` with `false` ?

Reply:
Tbh, I just based this off the current implementation in the SDKs...

The `1` mapping to `true` and `0` to `false` sounds reasonable though, especially because then the zero value for integers would be cast to `false`.

Which do you think is better?

Reply:
I'm partial to allowing it, then everything can be implicitly cast to anything else - and I think you can then remove an error type.

@jskeet any concerns?

Reply:
I'm *okay* with it - personally I do like being clear about expressions in terms of types, but that's mostly in C# etc. I haven't done significant amounts of SQL in years, so I don't know how it would "feel". If others are happy, that's fine by me.

Reply:
@duglin if we do this:
> map `1` with `true` and `0` with `false`

then how would we handle an integer value like `200`? Would that be an error? Or to avoid errors would it be better to say that when casting from integer to boolean, any non-zero integer becomes `true`, and `0` becomes `false`.

Reply:
Yup- I like that. Int->bool is zero vs non-zero.  But bool->int is 0 vs 1